Makefile

```diff
@@ -129,7 +131,10 @@ ifeq ($(UNICORN_SHARED),yes)
 ifeq ($(IS_MINGW),1)
 LIBRARY = $(BLDIR)/$(LIBNAME).$(EXT)
 else ifeq ($(IS_CYGWIN),1)
-LIBRARY = $(BLDIR)/$(LIBNAME).$(EXT)
+LIBRARY = $(BLDIR)/cyg$(LIBNAME).$(EXT)
```

Comment:
Cygwin's shared libraries are named `cyg*.dll` instead of `lib*.dll`.
https://cygwin.com/cygwin-ug-net/dll.html
> The name of your library is ${module}, prefixed with cyg for the DLL and lib for the import library. Cygwin DLLs use the cyg prefix to differentiate them from native-Windows MinGW DLLs, ...

And, Cygwin's dlopen(3) treats the file name `lib*.so` as `cyg*.dll`.
https://cygwin.com/git/gitweb.cgi?p=newlib-cygwin.git;a=blob;f=winsup/cygwin/dlfcn.cc
get_full_path_of_dll() converts the file name.